### PR TITLE
[20] API endpoint to get post details

### PR DESCRIPTION
**Pull request by moraesagustina**

### CODE REVIEW

File: app/views/api/v1/posts/show.json.jbuilder [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+json.call(@post, :id, :title, :body, :published_at, :user_id)
+
+json.likes @post.likes do |like|
+  json.user_id like.user_id
+  user = User.find(like.user_id)
+  json.nickname user.nickname
```
Comment (EnzoCozza) [inline app/views/api/v1/posts/show.json.jbuilder]: Like belongs to user, so you can just access it using `like.user`
```suggestion
  json.nickname like.user.nickname
```

File: spec/requests/api/v1/posts_request_spec.rb [hunk 2]
```diff
@@ -85,4 +85,79 @@
...
+        it 'contains expected attributes in the JSON response' do
+          json_response = JSON.parse(response.body)
+          expect(json_response.keys).to match_array(%w[id title body published_at user_id
+                                                       likes comments])
+          expect(json_response['likes'].first.keys).to match_array(%w[user_id nickname])
+          expect(json_response['comments'].first.keys).to match_array(%w[id content replies])
+        end
+      end
+
+      context 'when the user in the request does not exist' do
```
Comment (EnzoCozza) [inline spec/requests/api/v1/posts_request_spec.rb]:
```suggestion
      context 'when the post in the request does not exist' do
```

File: app/views/api/v1/posts/show.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+end
+
+json.comments @post.comments do |comment|
+  json.extract! comment, :id, :content
+
+  json.replies comment.comments do |reply|
```
Comment (EnzoCozza) [inline app/views/api/v1/posts/show.json.jbuilder]: This finishes in the first level of replies. If you also have a reply for that reply, it won't be shown. Let's make this recursive using partials!

File: README.md [hunk 4]
```diff
@@ -121,3 +121,79 @@ Returns all posts of a given user. This endpoint is accessible only to authentic
...
 }
 ```
+
+## Get Post Details Endpoint
+
+### Endpoint
+`GET /api/v1/posts/:id`
+
+### Description
+Returns the detals of a post. This endpoint is accessible only to authenticated users.
```
Comment (EnzoCozza) [inline README.md]:
```suggestion
Returns the details of a post. This endpoint is accessible only to authenticated users.
```

**Review by lucianocaus — APPROVED** [review]

**Pull request merged**